Support/lib/executable.rb

```diff
@@ -0,0 +1,93 @@
+require 'shellwords'
+
+module Executable
+  class NotFound < RuntimeError; end
+
+  class << self
+    # Try to find the Ruby executable `name`. Return an array containing the
+    # command(s) needed to run the executable. If the executable could not be
+    # found, `Executable::NotFound` is raised with a message describing the
+    # problem.
+    #
+    # Typical usage looks like this:
+    #
+    #     Dir.chdir(ENV['TM_PROJECT_DIRECTORY']) if ENV['TM_PROJECT_DIRECTORY']
+    #     begin
+    #       executable = Executable.find('rspec')
+    #       system(*executable, '--some', '--additional', 'args')
+    #     rescue Executable::NotFound => e
+    #       # Executable not found, so use fallback / display alert / …
+    #       # `e.message` contains detailled error message.
+    #     end
+    #
+    # Supports the following cases:
+    #
+    # 1. If an appropriate `TM_*` environment variable is present and points to
+    #    an executable file, its value is returned. (The name of the environment
+    #    variable is automatically derived from the executable name, eg. `rspec`
+    #    → `TM_RSPEC` etc. Alternatively, you can use the `env_var` argument
+    #    to explicitly specify the name.)
+    # 2. If a binstub (`bin/name`) exists, it is returned.
+    # 3. If a Gemfile.lock exists and has an entry for `name`, `[bundle exec
+    #    name]` is returned.
+    # 4. If `name` is found in the search path, it is returned. (Special case:
+    #    If `name` looks like an rbenv shim, also check if the executable has
+    #    been installed for the current Ruby version.)
+    #
+    # Both RVM and rbenv are supported, too:
+    #
+    # * If RVM is installed and the current directory contains an RVM project
+    #   file, the command to run the executable will be prefixed with `rvm .
+    #   do`. (Note that this is NOT the case if an appropriate `TM_*`
+    #   environment variable is present: In this case, the value of the
+    #   environment variable is returned unchanged. )
+    #
+    # * rbenv just works out of the box as long as your PATH (inside TextMate)
+    #   is setup to contain `~/.rbenv/shims`.
+    #
+    def find(name, env_var = nil)
+      # Safeguard against invalid names so that we don’t need to care about
+      # shell escaping later on
+      raise ArgumentError, "Invalid characters found in '#{name}'" unless name =~ /\A[\w_-]+\z/
```

Comment:
I like to be conservative and explicitly forbid use cases I have not thought of (because haven’t thought of a use case also means that there are no tests for it 😄). But I’ll be happy to relax this check if an actual use case shows up.